libpod/container_internal.go

```diff
@@ -492,11 +493,16 @@ func (c *Container) setupStorage(ctx context.Context) error {
 		containerInfo, containerInfoErr = c.runtime.storageService.CreateContainerStorage(ctx, c.runtime.imageContext, c.config.RootfsImageName, c.config.RootfsImageID, c.config.Name, c.config.ID, options)
 
 		if !generateName || !errors.Is(containerInfoErr, storage.ErrDuplicateName) {
+			errDuplicateName = true
```

Comment:
This is not really correct, the logic breaks here if an explcit name was requested or for nay other error then ErrDuplicateName. Thus if the calls fails for any other reason your error becomes misleading.
Also the extra bool var is not really needed you can just do the errors.IS check below.